enacts/onset/maproom.py

```diff
@@ -608,7 +662,7 @@ def onset_tile(tz, tx, ty):
             dry_spell_length,
             0
         )
-        mymap_max = np.timedelta64((precip_tile["T"][-1] - precip_tile["T"][0]).values, 'D')
+        map_max = np.timedelta64((precip_tile["T"][-1] - precip_tile["T"][0]).values, 'D')
     else:
         onset_dates = calc.seasonal_onset_date(
```

Comment:
I always have trouble remembering how `onset_date` and `seasonal_onset_date` are related. Better naming might help with that (not that I have a good suggestion at the moment), but also I think you could avoid the issue and simplify `onset_tile` by only using `seasonal_onset_date`, even in the monit case. It doesn't hurt to call that with only a single season's data, right?

Refreshing my memory on `onset_date` and `seasonal_onset_date` led to the following reflection. You can address it outside of this PR if you want. Conceptually, I see `seasonal_onset_date` as `extract_season` followed by `onset_date`, and likewise `seasonal_cess_date` is `extract_season` followed by `cessation_date`. But that's not how it's currently written. Instead, there are duplicate implementations of `extract_season` inside `seasonal_onset_date` and `seasonal_cess _date`.

Reply:
After the current water balance PR is merged, the next step will be to deal with multiply years calculation using the single year water balance algo so far written. We had said that we would likely not go about the method used here. Because, one, I think you'd rather we deal with this by using 2 dimensions rather than grouping; and two, because in any case the grouping features imply reduction of the group, which we want here; but don't with water balance.

And so maybe we get back to this (and thus the other conversation above) when we are ready to do multiple years water balance and we can assess which solution(s) we want for all these cases?